badge_server/main.py

```diff
@@ -75,519 +88,235 @@ def _get_pair_status_for_packages(pkg_sets):
-        google_compat_res['py3']['status'],
-        dependency_res['status']
-    ):
-        status = 'UNKNOWN'
-    else:
-        status = 'CHECK_WARNING'
-
-    # Get the latest timestamp
-    self_ts = self_compat_res.get('timestamp', '')
-    google_ts = google_compat_res.get('timestamp', '')
-    dep_ts = dependency_res.get('timestamp', '')
-    ts_list = [self_ts, google_ts, dep_ts]
-    ts_list.sort(reverse=True)
-    timestamp = ts_list[0] if ts_list else ''
-
-    return status, timestamp, \
-        self_compat_res, google_compat_res, dependency_res
+    Args:
+        package_name: the name of the package to get outdated dependencies for.
 
+    Returns:
+        A dict containing the pair compatibility status and details for any
+        pair incompatibilities. The dict will be formatted like the following:
 
-@app.route('/')
-def greetings():
-    """This is for testing the server health."""
-    return 'hello world'
-
-
-@app.route('/one_badge_image')
-def one_badge_image():
-    """Generate the badge for all the checks."""
-    package_name = flask.request.args.get('package')
-    badge_name = flask.request.args.get('badge_name')
-    is_github = False
-
-    if badge_name is None:
-        badge_name = package_name
-
-    if 'github.com' in badge_name:
-        badge_name = badge_utils.GITHUB_HEAD_NAME
-        is_github = True
-
-    commit_number = badge_utils._calculate_commit_number(package_name)
-
-    force_run_check = flask.request.args.get('force_run_check')
-    # Remove the last '/' from the url root
-    url_prefix = flask.request.url_root[:-1]
-    # Call the url for each badge to run the checks. This will populate the
-    # individual caches, which are used to calculate the final image state.
-    # Self compatibility badge
-    requests.get(url_prefix + flask.url_for(
-        'self_compatibility_badge_image',
-        package=package_name,
-        force_run_check=force_run_check,
-        commit_number=commit_number))
-    # Google compatibility badge
-    requests.get(url_prefix + flask.url_for(
-        'google_compatibility_badge_image',
-        package=package_name,
-        force_run_check=force_run_check,
-        commit_number=commit_number))
-    # Self dependency badge
-    requests.get(url_prefix + flask.url_for(
-        'self_dependency_badge_image',
-        package=package_name,
-        force_run_check=force_run_check,
-        commit_number=commit_number))
-
-    status, timestamp, _, _, _ = _get_all_results_from_cache(
-        package_name, commit_number=commit_number)
-    color = badge_utils.STATUS_COLOR_MAPPING[status]
-
-    details_link = url_prefix + flask.url_for('one_badge_target',
-                                              package=package_name)
-
-    # Include the check timestamp for github head
-    if is_github and timestamp:
-        badge_name = '{} {}'.format(badge_name, timestamp)
+        {
+            'py2': { 'status': 'SUCCESS', 'details': {} },
```

Comment:
I don't think that this is true. `details` seems to be a nested dict.

Reply:
updated